React/Views/RCTNavItem.m

```diff
@@ -11,10 +11,23 @@
 
 @implementation RCTNavItem
 
+@synthesize titleImageView = _titleImageView;
 @synthesize backButtonItem = _backButtonItem;
 @synthesize leftButtonItem = _leftButtonItem;
 @synthesize rightButtonItem = _rightButtonItem;
 
+- (UIImageView *)titleImageView
+{
+  if (!_titleImageView) {
+    if (_titleImage) {
```

Comment:
Thanks for jumping in @javache!

Yes, you're right, I reproduced the issue using `replacePreviousAndPop` with a previous route with an image and it doesn't get updated. I'll update the PR.